src/processor/Package.ts

```diff
@@ -83,4 +86,42 @@ export class Package {
       pullAt(sd.rules, rulesToRemove);
     });
   }
+
+  // Choice elements have a standard set of slicing rules applied to them by SUSHI.
+  // Therefore, it is not necessary to define that slicing using FSH when one of the choices exists.
+  // If the full set of four default rules exists for the same element, remove those rules.
+  private suppressChoiceSlicingRules(): void {
+    [...this.profiles, ...this.extensions].forEach(sd => {
+      const rulesToMaybeRemove: number[] = [];
+      sd.rules.forEach((rule, i, allRules) => {
+        if (rule instanceof ExportableCaretValueRule && rule.path.endsWith('[x]')) {
+          const pathStart = rule.path.replace(/\[x\]$/, '');
+          // check the four relevant caret paths and their default values, and
+          // check if one of the choices exists
+          if (
+            ((rule.caretPath === 'slicing.discriminator[0].type' &&
+              rule.value instanceof fshtypes.FshCode &&
+              rule.value.code === 'type') ||
+              (rule.caretPath === 'slicing.discriminator[0].path' && rule.value === '$this') ||
+              (rule.caretPath === 'slicing.ordered' && rule.value === false) ||
+              (rule.caretPath === 'slicing.rules' &&
+                rule.value instanceof fshtypes.FshCode &&
+                rule.value.code === 'open')) &&
+            allRules.some(
+              otherRule => otherRule.path != rule.path && otherRule.path.startsWith(pathStart)
```

Comment:
This line is checking that some choice is defined right? So something like `valueString` if the `[x]` element is `value[x]`? If so, I'm wondering if this is a strict enough check. What if the other rule is just something like `value[x] MS`. Wouldn't that  also trigger this case?

Reply:
In that case, `otherRule` and `rule` have the same path, so the condition would not be satisfied.

Reply:
That's true, I am still a little worried by this though, because doesn't it rely on the idea that no other element starts with the same path as the choice rule without `[x]`. I know on SUSHI we've gotten into trouble with two elements with out logic having bugs when one element's name is contained within that of another. Although maybe it is impossible to have an element that starts with the same string as the choice element, because then that element would look like a specific choice. So I think this is most likely safe, but I'm just not 100% sure.

Reply:
Hmm, thinking about it more carefully, I think you're right. We might be looking at `value[x]` and find another rule with a path of `value[x].id`, which satisfies this condition. So this does need to be tightened up.